sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpClient.cs

```diff
@@ -233,7 +250,7 @@ internal class AmqpClient : TransportClient
                         ++failedAttemptCount;
                         retryDelay = retryPolicy.CalculateRetryDelay(activeEx, failedAttemptCount);
 
-                        if ((retryDelay.HasValue) && (!ConnectionScope.IsDisposed) && (!cancellationToken.IsCancellationRequested))
+                        if ((retryDelay.HasValue) && (!ConnectionScope.IsDisposed) && (!_closed) && (!cancellationToken.IsCancellationRequested))
```

Comment:
This change seems unrelated to the custom endpoint. Is there a test covering it?

Reply:
They're orthogonal, but were overlooked earlier.   Unfortunately, we're deep enough in the AMQP space at this point that we can't unit test due to the lack of mocking support.    

In theory, this is redundant to the connection scope check, but there may be a slight delay for network operations there causing a (mostly) benign race.  The `_closed` check should not change the behavior but should give us a more deterministic detection.